programs/client/Client.cpp

```diff
@@ -1102,7 +1105,7 @@ void Client::processOptions(const OptionsDescription & options_description,
         config().setBool("no-warnings", true);
     if (options.count("fake-drop"))
         config().setString("ignore_drop_queries_probability", "1");
-    if (options.count("accept-invalid-certificate"))
+    if (options.count("accept-invalid-certificate") || config().has("accept-invalid-certificate"))
```

Comment:
this is useless - config file is not loaded at this stage yet - you need to do it in the `Client::initialize` function after config is loaded.